webapp/app/components/pages/ActionButtons/ActionButtons.jsx

```diff
@@ -93,41 +97,21 @@ export class ActionButtons extends React.Component {
     });
   };
 
-  validateForm = () => {
-    const { user } = this.state;
-    const isValidUsername = user.name !== '';
-    const isValidEmail = EMAIL_REGEXP.test(user.email);
-    const errors = {};
-
-    if (!isValidUsername) {
-      errors.name = 'User name is required';
-    }
-
-    if (!isValidEmail) {
-      errors.email = 'Email is invalid';
-    }
-
-    this.setState({ errors });
-
-    return errors;
-  };
-
-  canSubmitForm = () => {
-    const errors = this.validateForm();
-    return (Object.keys(errors).length === 0 && errors.constructor === Object);
-  };
-
   saveUser = () => {
-    if (!this.canSubmitForm()) {
-      return;
-    }
+    let validation = this.validator.validate(this.state.user);
 
-    if (typeof this.props.onConfirm === 'function') {
-      this.props.onConfirm(this.state.actionType, this.state.user)
-        .then(() => {
-          this.toggle();
-        });
-    }
+    this.setState({
+      validation
+    }, function () {
+      if (validation.isValid) {
+        if (typeof this.props.onConfirm === 'function') {
+          this.props.onConfirm(this.state.actionType, this.state.user)
+            .then(() => {
+              this.toggle();
+            });
+        }
+      }
```

Comment:
Move this callback to a method to simplfy this handler, and prefer use arrow callback.